koku/api/user_access/view.py

```diff
@@ -28,121 +28,88 @@ class UIFeatureAccess:
         1. to keep UI and API in sync about RBAC permissions
         2. to provide the UI (both our UI and the platform's chroming) with a simple boolean response regarding whether
             a user has access to specific features of the UI.
-
-    Class attributes:
-
-        access_keys (list) - a list of access keys to check. must be implemented by sub-classes.
-                             multiple keys will be ORed; i.e. if a user has access to anything in the list,
-                             this API returns True
     """
 
-    def __init__(self, access):
+    PRERELEASE = ["ibm"]
+
+    def __init__(self, access, admin_user, access_key_list=None):
         """Class Constructor.
 
         Args:
             access (dict) - an RBAC dict; see: koku.koku.middleware.IdentityHeaderMiddleware
 
         """
+        if not access_key_list:
+            self.access_key_list = self.access_key_mapping.keys()
         self.access_dict = access if access else {}
+        self.admin_user = admin_user
+        self.access_key_list = []
+
+    def set_access_key_list(self, query_param):
+        """Checks the access key list."""
+        if query_param:
+            if query_param in self.access_key_mapping:
+                self.access_key_list = [query_param]
+        else:
+            self.access_key_list = self.access_key_mapping.keys()
+        return bool(self.access_key_list)
+
+    @property
+    def access_key_mapping(self):
+        mapping = {}
+        mapping["aws"] = ["aws.account", "aws.organizational_unit"]
+        mapping["ocp"] = ["openshift.cluster", "openshift.node", "openshift.project"]
+        mapping["azure"] = ["azure.subscription_guid"]
+        mapping["gcp"] = ["gcp.account", "gcp.project"]
+        mapping["oci"] = ["oci.payer_tenant_id"]
+        mapping["ibm"] = ["ibm.account"]
+        mapping["cost_model"] = ["cost_model"]
+        mapping["settings"] = ["settings"]
```

Comment:
```suggestion
        mapping = {
            "aws": ["aws.account", "aws.organizational_unit"],
            "azure": ["azure.subscription_guid"],
            "gcp": ["gcp.account", "gcp.project"],
            "ibm": ["ibm.account"],
            "oci": ["oci.payer_tenant_id"],
            "ocp": ["openshift.cluster", "openshift.node", "openshift.project"],
            "cost_model": ["cost_model"],
            "settings": ["settings"],
        }
        mapping["any"] = mapping["aws"] + mapping["azure"] + mapping["gcp"] + mapping["oci"] + mapping["ocp"]
        return mapping

```

Reply:
It could also be worth making this dict a constant in this file. As a class property, this thing has to be recreated every single time the property is accessed which seems unnecessary.